### PR TITLE
CRUD

### PR DESCRIPTION
**Pull request by emkwak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+
+```sh
+npm install
+```
+## API
+
+| GET               | POST               | PUT                      | DELETE            |
+| ----------------- | ------------------ | ------------------------ | ----------------- |
+| READ              | CREATE             | UPDATE                   | DELETE            |
+| /earnings/:ticker | /earnings/add      | /earnings/update/:ticker | /earnings/:ticker |
```
Comment (bellcd) [inline README.md]: Is there any particular reason POST and PUT are at different endpoints? Could the 4 different verbs be on the same endpoint?

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
+      ticker: '',
       name: '',
       earnings: [],
       updated: false
     };
   }
 
   componentDidMount() {
-    fetch(`http://ec2-3-133-140-78.us-east-2.compute.amazonaws.com:3000/earnings/${this.state.ticker}`)
+    const value = queryString.parse(window.location.search)
```
Comment (bellcd) [inline src/App.js]: Neat solution!

File: README.md (hunk 1 above)
Comment (emkwak) [reply README.md]: @bellcd Thanks, I updated this if you want to review it again before I merge it.

File: data/index.js [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 
 var db = mongoose.connection;
 db.on('error', console.error.bind(console, 'connection error:'));
 db.once('open', () => console.log('db connected'));
 
 
 const earningsSchema = new mongoose.Schema({
-  ticker: String,
-  name: String,
+  ticker: { type: String, required: true, unique: true },
```
[the remark this answers is not in the document]
Comment (emkwak) [reply data/index.js]: @manuelag19 Right, having required set to true you must have a value in those fields. For unique set to true was to handle any duplicate entries. It logs back an error when you try to save duplicates.

File: server/app.js [hunk 4]
```diff
@@ -25,4 +25,35 @@ app.get("/earnings/:ticker", function(req, res) {
   });
 })
 
+// POST
+app.post('/earnings', (req, res) => {
+  const newEarning = new db.Earnings(req.body)
+
+  newEarning.save()
+    .then(() => res.json('Earning Added'))
+    .catch(err => res.status(400).json(`Error: ${err}`));
```
[the remark this answers is not in the document]
Comment (emkwak) [reply server/app.js]: client

File: server/app.js [hunk 5]
```diff
@@ -25,4 +25,35 @@ app.get("/earnings/:ticker", function(req, res) {
...
+        .then(() => res.json('Earning Updated'))
+        .catch(err => res.status(404).json(`Error: ${err}`));
+    })
+    .catch(err => res.status(404).json(`Error: ${err}`));
+})
+
+// DELETE
+app.delete('/earnings/:ticker', (req, res) => {
+  db.Earnings.findOneAndDelete({ ticker: req.params.ticker })
+    .then(() => res.json('Earning Deleted'))
```
[the remark this answers is not in the document]
Comment (emkwak) [reply server/app.js]: I set the status to 200

**Comment by emkwak** [comment]
@manuelag19  for duplicate entries it just throws an error. When running the seed script I need to figure out a way to just drop the duplicate and continue generating tickers.
For now setting random alphanumerics seems to be fine.

File: data/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (manuelag19) [reply data/index.js]: Sweet! What a nice way to handle duplicates

File: src/App.js [hunk 6]
```diff
@@ -28,11 +29,9 @@ class App extends Component{
             earnings: result.earnings,
             updated: true
           });
-        },
-        (error) => {
-          console.log(error)
         }
       )
+      .catch(err => console.log(`Error: ${err}`))
```
Comment (manuelag19) [inline src/App.js]: Nice!

**Pull request merged**